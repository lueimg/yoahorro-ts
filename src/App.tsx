import * as React from "react";
import Main from "./modules/Main";
import { Provider } from "react-redux";
import store from "./store/store";
import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom";
import Login from "./modules/auth/Login";
import NoMatch from "./shared/ui/NoMatch";
import PrivateSection from "./shared/ui/PrivateRoute";
import Logout from './modules/auth/Logout';

import "./App.css";
import "./assets/css/flex.css"
import "./assets/css/helpers.css"

class App extends React.Component {
  redirectTo404 = () => {
    return <Redirect to="/404" />
  }
  componentDidMount () {
    // test purpose
  }

  render() {
    return (
      <BrowserRouter>
        <Provider store={store}>
          <Switch>
            <Route exact path="/login" component={Login} />
            <Route path="/404" component={NoMatch} />
            <PrivateSection>
              <Route exact path="/logout" component={Logout} />
              <Route path="/" component={Main} />
            </PrivateSection>
          </Switch>
        </Provider>
      </BrowserRouter>
    );
  }
}

export default App;
