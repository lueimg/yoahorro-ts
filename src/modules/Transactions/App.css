.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.App-form-data {
  display: flex;
  flex-direction: column;
  width: fit-content;
  height: 300px;
  justify-content: space-around
}

.App-form-data-container {
  flex-direction: column;
  width: 300px;
  justify-content: center;
  display: flex;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.App-list {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
}

.App-list-item {
  width: 1000px;
  display: flex;
  flex-direction: row;
}

.App-list-item-left {
  margin: 10px;
  flex-direction: row;
}

.App-list-item-right {
  margin: 10px;
  flex-direction: row;
}

.App-list-item-account-transaction {
  display: flex;
  flex-direction: row
}

.Dashboard {
  justify-content: center;
  display: flex;
}

.Dashboard-square {
  justify-content: center;
  align-items: center;
  display: flex;
  padding-top: 50px;
  flex-direction: column;
}

.center {
  justify-content: center;
  display: flex;
}

.Dashboard-main-square {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 40%;
}