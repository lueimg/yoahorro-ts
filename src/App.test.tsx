import * as React from 'react';
import App from './App';
import { configure, shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter()})

describe('App', () => {
  it('should render app', () => {
    const wrapper = shallow(<App />, { context: {}, disableLifecycleMethods: true});
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})