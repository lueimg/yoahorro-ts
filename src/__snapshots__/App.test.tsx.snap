// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app 1`] = `
<BrowserRouter>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/404"
      />
      <Connect(PrivateRoute)>
        <Route
          component={[Function]}
          exact={true}
          path="/logout"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Connect(PrivateRoute)>
    </Switch>
  </Provider>
</BrowserRouter>
`;
